x-pack/plugins/observability/public/pages/slo_edit/components/slo_edit_form.tsx

```diff
@@ -67,44 +66,56 @@ export function SloEditForm({ slo }: Props) {
       watch,
     });
 
-  const { loading, success, error, createSlo, updateSlo } = useCreateOrUpdateSlo();
+  const { createSlo, updateSlo } = useCreateOrUpdateSlo();
+
+  const isEditMode = slo !== undefined;
 
-  const handleSubmit = () => {
+  const handleSubmit = async () => {
     const values = getValues();
+
     if (isEditMode) {
-      const processedValues = transformValuesToUpdateSLOInput(values);
-      updateSlo(slo.id, processedValues);
+      try {
+        const processedValues = transformValuesToUpdateSLOInput(values);
+
+        await updateSlo.mutateAsync({ sloId: slo.id, slo: processedValues });
+
+        toasts.addSuccess(
+          i18n.translate('xpack.observability.slos.sloEdit.update.success', {
+            defaultMessage: 'Successfully updated {name}',
+            values: { name: getValues().name },
+          })
+        );
```

Comment:
👍🏻  Pfew, that's nice. No more useEffect elsewhere to handle the success/error cases.